### PR TITLE
HMSPROV-157: Guard Pubkey DAO by tenant

### PR DESCRIPTION
**Pull request by ezr-ondrej**
Fixes HMSPROV-157

TODO: do the same for stubbed dao, once #102 is in.

### CODE REVIEW

**Comment by adiabramovitch** [comment]
looks great!
Do you add the same for the stubbed dao in this PR? 
I merged #102

**Comment by ezr-ondrej** [comment]
> Do you add the same for the stubbed dao in this PR?

Done :)

**Comment by ezr-ondrej** [comment]
I'm hitting an issue in tests, where I do not have account in context, which should have been solved when I've been working on the account, but it was not, so #108

**Comment by ezr-ondrej** [comment]
The test failure brings up interesting points, that I want to address:

- Should we filter out the accountID from the request? I'd propose yes in this PR
- Should we fill the nil AccountID on creation by the current tenant? I'll also propose yes

Converting to draft until those are addressed.

File: internal/dao/stubs/pubkey_dao.go [hunk 1]
```diff
@@ -28,39 +28,54 @@ func getPubkeyDao(ctx context.Context) (dao.PubkeyDao, error) {
...
+	for _, pk := range stub.store {
+		if pk.AccountID == ctxAccountId(ctx) && pk.ID == id {
+			return pk, nil
 		}
 	}
 	return nil, NewRecordNotFoundError(ctx, "Pubkey")
 }
 
 func (stub *pubkeyDaoStub) List(ctx context.Context, limit, offset int64) ([]*models.Pubkey, error) {
+	// not tenant filtered, should it?
```
Comment (adiabramovitch) [inline internal/dao/stubs/pubkey_dao.go:72]: I guess it should but to which account id do we compare?

**Review by adiabramovitch — APPROVED** [review]
except for the question I dropped, LGTM :)

File: internal/dao/stubs/pubkey_dao.go (hunk 1 above)
Comment (ezr-ondrej) [reply internal/dao/stubs/pubkey_dao.go:72]: I'll add it, it is just me being lazy... lets not be lazy :stuck_out_tongue:

Comment (ezr-ondrej) [reply internal/dao/stubs/pubkey_dao.go]: Alright, done :+1:

**Pull request merged**